hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DeploymentManagementTest.java

```diff
@@ -545,6 +534,44 @@ public void assignDistributionSetAndAutoCloseActiveActions() {
         }
     }
 
+    @Test
+    @Description("Verifies that if multi assignment is enabled previous Distribution is not canceled when a new one is assigned.")
+    @ExpectEvents({ @Expect(type = TargetCreatedEvent.class, count = 10),
+            @Expect(type = TargetUpdatedEvent.class, count = 20), @Expect(type = ActionCreatedEvent.class, count = 20),
+            @Expect(type = DistributionSetCreatedEvent.class, count = 2),
+            @Expect(type = SoftwareModuleCreatedEvent.class, count = 6),
+            @Expect(type = MultiActionEvent.class, count = 2),
+            @Expect(type = TargetAssignDistributionSetEvent.class, count = 0) })
+    public void previousAssignmentsAreNotCanceledInMultiAssignMode() {
+        setMultiAssignmentsEnabled(true);
+        final List<Target> targets = testdataFactory.createTargets(10);
+
+        // First assignment
+        final DistributionSet ds1 = testdataFactory.createDistributionSet("Multi-assign-1");
+        assignDistributionSet(ds1, targets);
+
+        assertDsExclusivelyAssignedToTargets(targets, ds1.getId(), true, Status.RUNNING);
```

Comment:
Maybe instead of `true`or `false` you could use a constant from type `boolean` so test is more readable:

```java
// Constant on at top of class
private static final boolean STATE_ACTIVE = true;
// ...
assertDsExclusivelyAssignedToTargets(targets, ds1.getId(), STATE_ACTIVE, Status.RUNNING)
```